app/src/ui/check-runs/ci-check-run-actions-logs.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react'
+import {
+  getFormattedCheckRunDuration,
+  IRefCheckOutput,
+  RefCheckOutputType,
+} from '../../lib/stores/commit-status-store'
+
+import { Octicon } from '../octicons'
+import classNames from 'classnames'
+import * as OcticonSymbol from '../octicons/octicons.generated'
+import {
+  getClassNameForLogStep,
+  getSymbolForLogStep,
+} from '../branches/ci-status'
+
+interface ICICheckRunActionLogsProps {
+  /** The check run to display **/
+  readonly output: IRefCheckOutput
+
+  /** Whether call for actions logs is pending */
+  readonly loadingLogs: boolean
+}
+
+interface ICICheckRunActionLogsState {
+  readonly stepOpenState: Map<number, boolean>
```

Comment:
First off I'd like it if we could make this a readonly map to ensure we don't accidentally mutate the map instead of replacing it? (as mutating won't cause us to rerender)

But, after thinking about it some more a `Map<T, boolean>` can usually be represented equally as well if not better with a `Set<T>`. In this case you only care about which sections _are open_, not which are explicitly closed.

```suggestion
  readonly openSections: ReadonlySet<number>
```